### ISSUE TITLE
Delayed Jobs

### ISSUE THREAD

**Issue by alexsbromberg** [pachyderm/pachyderm#842]
Hey guys, having some more issues with downstream commits getting triggered. I've got the following pipeline set up:

A -> B -> C -> D
and
E -> C

So everything has one input, except C, which has E and B as inputs. Everything has parallelism set to 1. Here's how the incremental flags are set up

| Pipeline | Input 1 | Incremental Flag 1 | Input 2 | Incremental Flag 2 |
| --- | --- | --- | --- | --- |
| B | A | DIFF |  |  |
| C | B | DIFF | E | DIFF |
| D | C | NONE |  |  |

Here's my issue:
1. When I first add stuff, and make my first commits to A and E, everything works fine. 
2. When I make another commit to E, C gets triggered and does what I expected. However D doesn't get triggered
3. Then I make a commit to A. B and C both get triggered as expected. D also gets triggered, but it does what i would have expected it to do after step 2. No job gets triggered for D as a result of the commit to A. 

I also tried reversing steps two and 3, and get the same issue (in reverse order). Seems like there's some sort of issue of delayed scheduling.

Also potentially relevant, I'm experience the CrashLoopBackOff issue mentioned in issue [#837](https://github.com/pachyderm/pachyderm/issues/837)

I'm running the following version of pachctl / pachd:

```
COMPONENT           VERSION                                          
pachctl             1.2.0-399ffcb287a970e65c901ff0ee3288cad173f512   
pachd               1.2.0-399ffcb287a970e65c901ff0ee3288cad173f512 
```

As an aside, what setup do you guys use to run your tests? (are there additional dependencies required, etc.) I was going to write a test to replicate this behavior, but couldn't get the test suite working.

**Comment by derekchiang** [comment]
There have been 4 point releases since this issue was opened, during which a lot of PPS bugs were fixed.  Gonna close for now unless this issue can still be reproduced.

**Comment by jdoliner** [comment]
More importantly, the test that @alexsbromberg added to reproduce this issue now passes and is part of CI. So closing seems quite reasonable to me.

**Comment by derekchiang** [comment]
@jdoliner actually the test was never enabled.  I just enabled it and turns out it still won't pass.  Reopening.

**Comment by jdoliner** [comment]
@derekchiang good catch

**Comment by jdoliner** [comment]
Looks like the PR that fixes this has failing CI right now. It'd be nice to get this fixed in time for 1.3. @derekchiang do you think that's possible?
